compiler_opt/tools/bc_hyperparam_tuning.py

```diff
@@ -0,0 +1,92 @@
+# coding=utf-8
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script for testing different hyperparameter combinations for BC training."""
+
+import os
+
+from absl import app
+from absl import flags
+import tempfile
+import gin
+
+from compiler_opt.rl import train_bc
+from compiler_opt.tools import generate_default_trace
+
+import tensorflow as tf
+from tensorboard.plugins.hparams import api as hp
+
+_DATA_PATH = flags.DEFINE_multi_string(
+    'data_path', [],
+    'Path to TFRecord file(s) containing training data. Skip training and dump'
+    'an untrained model with random weights (for testing purpose) if '
+    'unspecified.')
+_CORPUS_PATH = flags.DEFINE_string(
+    'corpus_path', '', 'Path to the corpus to evaluate the models.')
+_OUTPUT_DIR = flags.DEFINE_string('output_dir', '', 'Path to the output data')
+_GIN_FILES = flags.DEFINE_multi_string(
+    'gin_files', [], 'List of paths to gin configuration files.')
+_GIN_BINDINGS = flags.DEFINE_multi_string(
+    'gin_bindings', [],
+    'Gin bindings to override the values set in the config files.')
+
+METRIC_REWARD = 'reward'
+
+
+def run_training():
+  with tempfile.TemporaryDirectory() as temp_dir:
+    root_dir = os.path.expanduser(temp_dir)
+    root_dir = os.path.normpath(root_dir)
+    # TODO(boomanaiden154): leave num_iterations and batch_size to config
```

Comment:
i guess they can be configured with gin file?

Reply:
That should definitely be possible. I'll have to figure out exactly how to get it working (and the exact syntax we want), but it shouldn't be too difficult.

Reply:
Switched to configuring the hyperparams to configure with a new gin config file, `compiler_opt/rl/reglloc/gin_configs/bc_hyperparam_tuning.gin`.